pkg/sharing/files.go

```diff
@@ -648,7 +651,10 @@ func (s *Sharing) CreateDir(inst *instance.Instance, target map[string]interface
 		return err
 	}
 	dir.SetID(target["_id"].(string))
+	ref.SID = consts.Files + "/" + dir.DocID
 	copySafeFieldsToDir(target, dir)
+	_, ruleIndex := s.findRuleForNewDirectory(dir)
```

Comment:
We might want to check if the rule is `ǹil` to make sure the ruleIndex is legit

Reply:
Good catch!